kite-data/kite-data-oozie/src/main/java/org/kitesdk/data/oozie/KiteURIHandler.java

```diff
@@ -101,9 +107,35 @@ public Context getContext(final URI uri, final Configuration conf,
     return null;
   }
 
+  private void loadConfiguration() {
+    if(Services.get() != null) {
+      HCatAccessorService hcatService = Services.get().get(HCatAccessorService.class);
+      if(hcatService != null) {
+        Configuration hcatConf = hcatService.getHCatConf();
+        if(hcatConf != null) {
+          DefaultConfiguration.set(hcatConf);
+        } else {
+          // kite conf was null
+          LOG.warn("Configuration for Kite not loaded, HCatAccessorService config was null.");
+        }
+      } else {
+        // service was null
+        LOG.error("Configuration for Kite not loaded, HCatAccessorService was not available.");
+      }
+    } else {
+      // services were null
+      LOG.error("Configuration for Kite not loaded, oozie services were not available.");
+    }
+  }
+
   @SuppressWarnings("rawtypes")
   @Override
   public boolean exists(final URI uri, final Context context) throws URIHandlerException {
+    // load the configuration if it hasn't been yet
+    if(confLoaded.compareAndSet(false, true)) {
+      loadConfiguration();
+    }
```

Comment:
It's worth mentioning that I load the configuration here in the exists method instead of the init method. In my testing I found that Oozie would initialize the URIHandlers before it initialized the services and the desired services would not be available.